### ISSUE TITLE
Failed drag to place cards

### ISSUE THREAD

**Issue by NomadCHK** [AdriaGual/pokemon-pocket-bot#19]
![image](https://github.com/user-attachments/assets/46ab49e3-3ff9-4e46-8f69-b26a6a601e33)
![image](https://github.com/user-attachments/assets/4838bda7-5730-4e19-9136-53f578927cf6)

This screenshot is an example. Maybe it's the starting position, or the timing. Or the problem could be the straight motion, maybe we need to add some curve.

It would be nice if someone could join me in figuring this out. I've added a new debug window to the refactor branch of the fork repository. #18

**Comment by AdriaGual** [comment]
I’ve noticed that the issue seems to be related to both speed and distance. Let me explain:

Depending on the card’s position, moving it to a bench position can require either a longer or shorter distance. During testing, I observed that cards with a longer distance to travel were more likely to be successfully played.

I think we should consider an approach that takes both factors into account: the card’s position and the position of the bench.

Hope this helps!

**Comment by TerpToke** [comment]
@NomadCHK @AdriaGual My knowledge is limited but I'm very tech savvy and adore learning these kind of things. The learning aspect comes easy. If there's ever anything I can do to help this process, let me know!

**Comment by NomadCHK** [comment]
> @TerpToke You can help in a lot of ways if you want, this days I felt like people would appreciate having a discord server to share knowledge, like a FAQ so people can start using the bot it easily, maybe also an instalation tutorial would be nice.
> 
> Another thing that came to my mind is taking captures of the same images that we use in english but for other languages, that would help moving a step further to the multilanguage bot.
> 
> If you wanna code, I would tell you to wait a bit, not for anything but @NomadCHK is working on an amazing refactor of the whole code structure #18
> 
> This refactor changes a lot of stuff (it's an amazing work), but I feel it will be easier for you when all this is merged!
> 
> Also you can contribute with ideas (stuff you would like to have on the bot)
> 
> All help is super helpful ❤️

About this, I think it's easy to change the language to English, and English is very common among most users. My ideas are to focus on integrating it on more devices (I need to improve the similarity calculation to handle different resolutions) and also automatically scale the coordinates that are encoded. I've already done some tests with this and managed to run it on my phone, I had some problems finding some images like the timer in a different resolution, but I think it's possible to do.

## But before we do that, maybe we need to make the game stable with the following:

Game state tracking: make it count energy and track the field better (less redundancies, track hp, points, etc.).
Improve actions: confirm actions (maybe we can use/read the battle log), curved swipe (drag on multiple points to improve card positioning).
And then finally:
AI (decision tree or others): start making the bot smart, with it playing by the rules and some strategies.

I think the most hard now is to make game state tracking work well

**Comment by NomadCHK** [comment]
So, just to make a small update, the latest changes have improved a bit, but the problem still persists when trying to place a card that is too far to the right on the left bench.

**Comment by AdriaGual** [comment]
Yeah, this is the bug, as you say, it can't play the card from right to the left bench

![image](https://github.com/user-attachments/assets/2a2ac3e1-0184-40e4-bb37-90e8222d1318)

Also seems that tries to play misty as if it was a pokémon to the bench?
![image](https://github.com/user-attachments/assets/b73c1a88-e3ca-4830-ad70-4d7e1d57e112)

**Comment by NomadCHK** [comment]
@AdriaGual take a look at this:
https://github.com/dtmilano/AndroidViewClient/blob/master/examples/helper/swipe

maybe the android view client can solve the dragging issues, that way we can do the Y movement first to place the cards. Let me know if you think it's worth switching from pure adb to this. I'll do some tests...
